tests/third_party/osmosis_testnet/net_config.py

```diff
@@ -18,42 +18,44 @@
 # ------------------------------------------------------------------------------
 
 """Osmosis network config."""
-from time import sleep
-
-import requests
-
+from cosmpy.aerial.client import LedgerClient
 from cosmpy.aerial.config import NetworkConfig
+from cosmpy.aerial.wallet import LocalWallet
+
 
+# Network config
+PREFIX = "osmo"
+COIN = "uosmo"
+GAS_LIMIT = 120000
 
 NET_CONFIG = NetworkConfig(
-    chain_id="osmo-test-4",
-    url="grpc+http://grpc-test.osmosis.zone:443/",
+    chain_id="osmo-test-5",
+    url="grpc+https://grpc.osmotest5.osmosis.zone:443",
     fee_minimum_gas_price=1,
-    fee_denomination="uosmo",
-    staking_denomination="uosmo",
+    fee_denomination=COIN,
+    staking_denomination=COIN,
 )
+
 DEFAULT_TIMEOUT = 60.0
+# PK for funded wallet osmo1ya7ymmhq3xq94parz46w5jhf2v83hqd83aac8e
+FUNDED_PK = "974c467d48351204b8481736079edfefa98eba16f4b80d74b279e103e8ee9246"
 
 
 class FaucetMixIn:
     """Osmosis faucet config"""
 
-    def ask_funds(self, wallet):
+    def ask_funds(
+        self, wallet: LocalWallet, ledger: LedgerClient, amount: int = 1000000
+    ):
         """Request fund from faucet.
 
         :param wallet: Wallet Address
+        :param ledger: LedgerClient
+        :param amount: amount to topup
         :raises Exception: fail to topup
         """
-        resp = requests.post(
-            "https://testnet-faucet.dev-osmosis.zone/request",
-            json={"address": str(wallet.address())},
-            timeout=DEFAULT_TIMEOUT,
-        )
-        assert resp.status_code == 200
-        ledger = self.get_ledger()
-        for i in range(10):
-            if ledger.query_bank_balance(wallet.address()) > 0:
-                break
-            sleep(i * 2)
-        else:
-            raise Exception("fail to topup")
+
+        funded_wallet = LocalWallet.from_unsafe_seed(FUNDED_PK, prefix=PREFIX)
+        ledger.send_tokens(
+            wallet.address(), amount, COIN, funded_wallet
+        ).wait_to_complete()
```

Comment:
Replaced facuet API which doesn't exist for new testnet with requesting funds from the well-funded account.